File: ejemplos/Golang/CURLpeticion.go
package main

import (
	"fmt"
	curl "github.com/IngenieroRicardo/curl/go"
)

func main() {
	var header string

	header = curl.Header("Accept", "application/json") + "\n" +
		curl.Header("Content-Type", "application/json") + "\n" +
		curl.HeaderAuthBasic("admin", "123456") + "\n"
	
	body := "{\"title\": \"foo\",\"body\": \"bar\",\"userId\": 1}"

	// Hacer petición Post
	resp := curl.Post("https://jsonplaceholder.typicode.com/posts", header, body)
    
	if resp != "" {
		fmt.Printf("Response:\n%s\n", resp)
	}
}

File: ejemplos/C/CURLpeticion.c
#include <stdio.h>
#include "curl.h"
#include "array.h"

// Compilar en Windows: gcc -o CURLpeticion.exe CURLpeticion.c ./curl.dll ./array.dll 
// Compilar en tipo UNIX: gcc -o CURLpeticion.bin CURLpeticion.c ./curl.so ./array.so 

int main() {
    String header = Concat(
                        Header("Accept", "application/json"), "\n",
                        Header("Content-Type", "application/json"), "\n",
                        HeaderAuthBasic("admin", "123456"), "\n",
                        NULL
                    );

    String body = Concat("{\"title\": \"foo\",\"body\": \"bar\",\"userId\": 1}", NULL);

    // Hacer petición Post
    String resp = Post("https://jsonplaceholder.typicode.com/posts", header, body);
    
    if (resp != NULL) {
        printf("Respuesta:\n%s\n", resp);
        FreeString(resp);
    }
    
    // Liberar memoria
    FreeString(header);
    FreeString(body);
    
    return 0;
}
